keras_nlp/samplers/sampler.py

```diff
@@ -164,6 +164,17 @@ def body(prompt, cache, index):
         )
         return prompt
 
+    def compute_probabilities(self, logits):
+        """Compute token probabilities from logits.
+
+        This will always be done in full precision, regardless of dtype, and
+        scale by `temperature`.
+        """
+        dtype = logits.dtype
```

Comment:
Maybe `logits_dtype`? (For consistency of style with `inputs_dtype` a few files up)